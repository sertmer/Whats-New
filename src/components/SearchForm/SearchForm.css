.nav {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 5vh;
}

h1 {
  display: flex;
  flex-direction: row;
  color: #ffffff;
  font-size: xx-large;
  padding: 10px;
}

span {
  color: #f08623;
}

.search-input {
  width: 80%;
  background-color: #141527;
  border: none;
  border-radius: 50px;
  font-size: larger;
  color: #7e7d7d;
  padding: 20px 20px;
  margin: 10px;
}

.search-submit {
  width: 10%;
  height: 5vh;
  border-radius: 50px;
  background-color: #7d48a8;
  border: 0px;
  color: #ffffff;
  font-size: larger;
  margin-top: 10px;
}