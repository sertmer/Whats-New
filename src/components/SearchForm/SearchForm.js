import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      search: '',
    }
  }
  
  searchCurrentPage = (query) => {
    const capitalQuery = query.toUpperCase()
    console.log(capitalQuery);
    this.state.currentPage.filter(story => {
      return story.title.contains(capitalQuery)
    })
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }

  render() { 
    return (
    <nav className='nav'>
      <h1>What's <span> New? </span></h1>
      <input 
        name='search'
        value={this.state.search}
        type='text' 
        placeholder='Search for news here'
        className='search-input'
        onChange={this.handleChange}>
      </input>
      <button className='search-submit' onClick={() => this.searchCurrentPage(this.state.search)}>Search</button>
    </nav>
  )}
}
export default SearchForm;