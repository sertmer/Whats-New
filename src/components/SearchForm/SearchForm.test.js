import React from 'react'
import SearchForm from './SearchForm'
import { shallow } from 'enzyme'

describe('SearchForm', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<SearchForm 
      searchCurrentPage={jest.fn()}
      />)
    
    expect(wrapper).toMatchSnapshot()
  })
})