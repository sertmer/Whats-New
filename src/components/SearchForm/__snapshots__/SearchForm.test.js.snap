// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data passed in correctly 1`] = `
<nav
  className="nav"
>
  <h1>
    What's 
    <span>
       New? 
    </span>
  </h1>
  <input
    className="search-input"
    name="search"
    onChange={[Function]}
    placeholder="Search for news here"
    type="text"
    value=""
  />
  <button
    className="search-submit"
    onClick={[Function]}
  >
    Search
  </button>
</nav>
`;
